File: frontend/src/stores/cari.ts
import { defineStore } from 'pinia'
import {
  getCarisApi,
  getCariSummaryApi,
  getCariByIDApi,
  createCariApi,
  updateCariApi,
  deleteCariApi,
  getCariFinancialSummaryApi,
  getCariTransactionsApi,
  addCariPersonApi,
  updateCariPersonApi,
  removeCariPersonApi
} from '@/api/cari'

export interface CariBalance {
  id: string
  cari_id: string
  currency: string
  balance: string
}

export interface CariPerson {
  id: string
  cari_id: string
  name: string
  title: string
  phone: string
  email: string
  created_at: string
}

export interface Cari {
  id: string
  company_id: string
  code: string
  type: string
  group?: string
  title: string
  name: string
  contact_name: string
  tax_office: string
  tax_number: string
  email: string
  phone: string
  landline: string
  fax: string
  address: string
  city: string
  district: string
  postal_code: string
  country: string
  shipping_address: string
  shipping_city: string
  shipping_district: string
  shipping_postal_code: string
  shipping_country: string
  currency: string
  opening_balance: string
  balances: CariBalance[]
  persons: CariPerson[]
  is_active: boolean
  note: string
}

export interface CariTransaction {
  id: string
  company_id: string
  cari_id: string
  date: string
  type: 'debit' | 'credit'
  source_type: 'invoice' | 'payment' | 'expense' | 'manual'
  source_id: string | null
  description: string
  amount: string
  balance_after: string
}

export const useCariStore = defineStore('cari', {
  state: () => ({
    caris: [] as Cari[],
    total: 0,
    summary: {
      total_receivables: [] as { currency: string; amount: string }[],
      total_payables: [] as { currency: string; amount: string }[],
      net_balance: [] as { currency: string; amount: string }[],
    },
    activeCari: null as Cari | null,
    activeCariFinancialSummary: null as any,
    activeCariTxs: [] as CariTransaction[],
    activeCariTxsTotal: 0,
    loading: false,
  }),
  actions: {
    async fetchCaris(params?: any) {
      this.loading = true
      try {
        const res = await getCarisApi(params)
        this.caris = res.data.data
        this.total = res.data.meta.total
      } finally {
        this.loading = false
      }
    },
    async fetchSummary() {
      const res = await getCariSummaryApi()
      this.summary = res.data.data
    },
    async fetchCariByID(id: string) {
      const res = await getCariByIDApi(id)
      this.activeCari = res.data.data
    },
    async fetchFinancialSummary(id: string) {
      const res = await getCariFinancialSummaryApi(id)
      this.activeCariFinancialSummary = res.data.data
    },
    async fetchTransactions(id: string, params?: any) {
      const res = await getCariTransactionsApi(id, params)
      this.activeCariTxs = res.data.data
      this.activeCariTxsTotal = res.data.meta.total
    },
    async createCari(data: any) {
      const res = await createCariApi(data)
      return res.data.data
    },
    async updateCari(id: string, data: any) {
      const res = await updateCariApi(id, data)
      return res.data.data
    },
    async deleteCari(id: string) {
      await deleteCariApi(id)
    },
    async addPerson(cariId: string, data: { name: string; title?: string; phone?: string; email?: string }) {
      const res = await addCariPersonApi(cariId, data)
      if (this.activeCari) {
        if (!this.activeCari.persons) this.activeCari.persons = []
        this.activeCari.persons.push(res.data.data)
      }
      return res.data.data
    },
    async updatePerson(cariId: string, personId: string, data: { name: string; title?: string; phone?: string; email?: string }) {
      const res = await updateCariPersonApi(cariId, personId, data)
      if (this.activeCari && this.activeCari.persons) {
        const idx = this.activeCari.persons.findIndex(p => p.id === personId)
        if (idx !== -1) this.activeCari.persons[idx] = res.data.data
      }
      return res.data.data
    },
    async removePerson(cariId: string, personId: string) {
      await removeCariPersonApi(cariId, personId)
      if (this.activeCari && this.activeCari.persons) {
        this.activeCari.persons = this.activeCari.persons.filter(p => p.id !== personId)
      }
    }
  },
})

File: backend/models/cari.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Cari struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey" json:"id"`
	CompanyID uuid.UUID `gorm:"type:uuid;not null;index" json:"company_id"`
	Code      string    `gorm:"type:varchar(100);not null" json:"code"`
	Type      string    `gorm:"type:varchar(50);not null" json:"type"` // 'customer', 'supplier', 'both', or custom
	Group     string    `gorm:"type:varchar(100)" json:"group"`
	Title     string    `gorm:"type:varchar(255)" json:"title"`        // Resmi ünvan (opsiyonel)
	Name      string    `gorm:"type:varchar(255)" json:"name"`         // Firma adı / cari adı
	ContactName string  `gorm:"type:varchar(255)" json:"contact_name"` // Ad Soyad / yetkili kişi
	TaxOffice string    `gorm:"type:varchar(255)" json:"tax_office"`
	TaxNumber string    `gorm:"type:varchar(50)" json:"tax_number"`
	Email     string    `gorm:"type:varchar(255)" json:"email"`
	Phone     string    `gorm:"type:varchar(50)" json:"phone"`
	Landline  string    `gorm:"type:varchar(50)" json:"landline"`
	Fax       string    `gorm:"type:varchar(50)" json:"fax"`

	// Fatura adresi
	Address    string `gorm:"type:text" json:"address"`
	City       string `gorm:"type:varchar(100)" json:"city"`     // İl
	District   string `gorm:"type:varchar(100)" json:"district"` // İlçe
	PostalCode string `gorm:"type:varchar(20)" json:"postal_code"`
	Country    string `gorm:"type:varchar(100);not null;default:'Türkiye'" json:"country"`

	// Sevk adresi
	ShippingAddress    string `gorm:"type:text" json:"shipping_address"`
	ShippingCity       string `gorm:"type:varchar(100)" json:"shipping_city"`
	ShippingDistrict   string `gorm:"type:varchar(100)" json:"shipping_district"`
	ShippingPostalCode string `gorm:"type:varchar(20)" json:"shipping_postal_code"`
	ShippingCountry    string `gorm:"type:varchar(100)" json:"shipping_country"`

	Currency  string        `gorm:"type:varchar(10);not null;default:'TRY'" json:"currency"`
	Balances  []CariBalance `gorm:"foreignKey:CariID" json:"balances"`
	Persons   []CariPerson  `gorm:"foreignKey:CariID" json:"persons"`
	IsActive  bool          `gorm:"not null;default:true" json:"is_active"`
	Note      string        `gorm:"type:text" json:"note"`
	CreatedBy *uuid.UUID    `gorm:"type:uuid" json:"created_by"`
	CreatedAt time.Time     `json:"created_at"`
	UpdatedAt time.Time     `json:"updated_at"`
}

type CariPerson struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey" json:"id"`
	CariID    uuid.UUID `gorm:"type:uuid;not null;index" json:"cari_id"`
	Name      string    `gorm:"type:varchar(255);not null" json:"name"`
	Title     string    `gorm:"type:varchar(100)" json:"title"`   // Unvan (Müdür, Muhasebe vb.)
	Phone     string    `gorm:"type:varchar(50)" json:"phone"`    // Kişi cep telefonu
	Email     string    `gorm:"type:varchar(255)" json:"email"`   // Kişi e-postası
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (CariPerson) TableName() string {
	return "cari_persons"
}
